app/src/main/java/com/github/multimatum_team/multimatum/repository/FirebaseGroupRepository.kt

```diff
@@ -109,12 +116,42 @@ class FirebaseGroupRepository @Inject constructor(database: FirebaseFirestore) :
     }
 
     /**
-     * Invite an user to join a group.
-     * @param id the ID of the grop to which we want to invite the user
-     * @param email the email of the user to invite
+     * Generate an invite link to join a group.
+     * @param id the ID of the group to which we want to invite users
      */
-    override suspend fun invite(id: GroupID, email: String) {
-        throw NotImplementedError("FirebaseGroupRepository.invite is not implemented")
+    override suspend fun generateInviteLink(id: GroupID): Uri {
+        val group = fetch(id)
+        val inviteLink = Uri.Builder()
+            .scheme("https")
+            .authority("multimatum.page.link")
+            .appendQueryParameter("id", group.id)
+            .build()
+
+        val dynamicLink = Firebase.dynamicLinks.dynamicLink {
+            link = inviteLink
+            domainUriPrefix = "https://multimatum.page.link"
```

Comment:
You could extract these string variables

Reply:
I left the link hardcoded for now but extracted the link title and description , I'm not sure exactly what kind of strings are expected to be string resources, I think string resources are mainly here to facilitate internationalization but I could be wrong.

Reply:
I resolved this problem (for the above comment too) by decoupling android dependencies from the repository, now the link title and description are passed as parameters.